xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-api/src/main/java/org/xwiki/attachment/internal/MoveAttachmentDocumentInitializer.java

```diff
@@ -0,0 +1,115 @@
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+import static org.xwiki.uiextension.internal.WikiUIExtensionConstants.CONTENT_PROPERTY;
+import static org.xwiki.uiextension.internal.WikiUIExtensionConstants.EXTENSION_POINT_ID_PROPERTY;
+import static org.xwiki.uiextension.internal.WikiUIExtensionConstants.ID_PROPERTY;
+import static org.xwiki.uiextension.internal.WikiUIExtensionConstants.SCOPE_PROPERTY;
+import static org.xwiki.uiextension.internal.WikiUIExtensionConstants.UI_EXTENSION_CLASS;
+
+/**
+ * Initializes a User Interface Extension (UIX) in a document. This UIX is used to initialize a configuration that can
+ * be used by Javascript scripts.
+ *
+ * @version $Id$
+ * @since 14.0RC1
+ */
+@Component
+@Singleton
+@Named("Attachment.Code.MoveAttachment")
+public class MoveAttachmentDocumentInitializer extends AbstractMandatoryDocumentInitializer
+{
+    private static final String UIPX_ID = "org.xwiki.platform.template.header.after";
+
+    private static final String UIX_ID = "org.xwiki.platform.attachment.header.after.move";
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private Logger logger;
+
+    /**
+     * Default constructor.
+     */
+    public MoveAttachmentDocumentInitializer()
+    {
+        super(new LocalDocumentReference(Arrays.asList("Attachment", "Code"), "MoveAttachment"));
+    }
+
+    @Override
+    public boolean updateDocument(XWikiDocument document)
+    {
+        List<BaseObject> uIExtensionObjects = document.getXObjects(UI_EXTENSION_CLASS);
+        boolean exists = uIExtensionObjects.stream().anyMatch(object ->
+            Objects.equals(object.getStringValue(EXTENSION_POINT_ID_PROPERTY), UIPX_ID)
+                && Objects.equals(object.getStringValue(ID_PROPERTY), UIX_ID));
+        if (!exists) {
+            try {
+                EntityReference uIExtensionClass = new EntityReference(UI_EXTENSION_CLASS);
+                int idx = document.createXObject(uIExtensionClass, this.contextProvider.get());
+                BaseObject xObject = document.getXObject(uIExtensionClass, idx);
+                xObject.setStringValue(EXTENSION_POINT_ID_PROPERTY, UIPX_ID);
+                xObject.setStringValue(ID_PROPERTY, UIX_ID);
+                xObject.setStringValue(CONTENT_PROPERTY, "{{velocity}}\n"
+                    + "{{html clean=\"false\"}}\n"
+                    + "<script id=\"attachment-move-config\" type=\"application/json\">$jsontool.serialize({\n"
+                    + "  'treeWebjar': $!services.webjars.url('org.xwiki.platform:xwiki-platform-tree-webjar', "
+                    + "'require-config.min.js', {'evaluate': true})"
+                    + "})</script>\n"
+                    + "{{/html}}\n"
+                    + "{{/velocity}}");
+                xObject.setStringValue(SCOPE_PROPERTY, "wiki");
+            } catch (XWikiException e) {
+                this.logger.error(
```

Comment:
We usually log with using the exception message only in such case. Moreover maybe to reuse the constants for the error log message.

Reply:
I'm not sure what the issue here. We have several instances of logging errors on document initializers following the same pattern (for instance, https://github.com/xwiki/xwiki-platform/blob/95062b970dc37c5529ebdbe19c47e38619a48176/xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-notifications/src/main/java/org/xwiki/mentions/internal/descriptors/NotificationAdministrationDocumentInitializer.java#L103)

Reply:
Nevermind mixing up the rule between warn messages and error messages.